src/openrct2/interface/Viewport.cpp

```diff
@@ -543,8 +543,8 @@ void viewport_update_position(rct_window* window)
 
     viewport_set_underground_flag(0, window, viewport);
 
-    auto viewportMidPoint = ScreenCoordsXY{ window->savedViewPos.x + viewport->view_width / 2,
-                                            window->savedViewPos.y + viewport->view_height / 2 };
+    auto viewportMidPoint = ScreenCoordsXY{ static_cast<int16_t>(window->savedViewPos.x + viewport->view_width / 2),
```

Comment:
Add a comment explaining why the casts are here.

Reply:
I thought about it and forgot, thanks for reminding. Do squash on merge, please